### PR TITLE
Nix fixes

### PR DESCRIPTION
**Pull request by puffnfresh**
This lets us build the Eta and start a shell. Trying to run `eta-build install` gives this:

```
$ eta-build install
# eta (for install)
# git (for install)# eta-pkg (for install)
# etlas (for install)
Downloading the latest package lists from:
- hackage.haskell.org
- git@github.com/typelead/etlas-index
Updating binary package index.
Verification loop. Errors in order:
  Version of <repo>/timestamp.json is less than the previous version
  Version of <repo>/timestamp.json is less than the previous version
  Version of <repo>/timestamp.json is less than the previous version
  Version of <repo>/timestamp.json is less than the previous version
  Version of <repo>/timestamp.json is less than the previous version
Error when running Shake build system:
* install
user error (Development.Shake.cmd, system command failed
Command: etlas update
Exit code: 1
Stderr:
Verification loop. Errors in order:
  Version of <repo>/timestamp.json is less than the previous version
  Version of <repo>/timestamp.json is less than the previous version
  Version of <repo>/timestamp.json is less than the previous version
  Version of <repo>/timestamp.json is less than the previous version
  Version of <repo>/timestamp.json is less than the previous version
)
```

### CODE REVIEW

**Comment by puffnfresh** [comment]
Maybe I just needed an `etlas update` cause now I get:

```
$ eta-build install
# eta (for install)
# git (for install)
# eta-pkg (for install)
# etlas (for install)
Downloading the latest package lists from:
- hackage.haskell.org
- git@github.com/typelead/etlas-index
To revert to previous state run:
    etlas update --index-state='2017-12-17T21:35:04Z'
Updating binary package index.
Updating the eta-hackage patch set.
# etlas (for install)
Will select the version of Eta available on the PATH.
# javac (for install)
# javac (for libraries/rts/build/rts.jar)
Note: eta/runtime/storage/ManagedHeap.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
# jar (for libraries/rts/build/rts.jar)
# etlas (for install)
Resolving dependencies...
Configuring rts-0.1.0.0...
Building rts-0.1.0.0...
Installed rts-0.1.0.0
# etlas (for install)
etlas: Error reading local package.
Couldn't find .cabal file in: .
Error when running Shake build system:
* install
user error (Development.Shake.cmd, system command failed
Command: etlas install --allow-boot-library-installs
Current directory: /home/brian/code/eta/libraries/eta-boot-th
Exit code: 1
Stderr:
etlas: Error reading local package.
Couldn't find .cabal file in: .
)
```

**Comment by rahulmutt** [comment]
Does wiping out `~/.etlas/packages` solve the problem?

**Comment by rahulmutt** [comment]
@puffnfresh You need to manually delete `libraries/eta-boot-th` folder locally and then run your build. Git doesn't remove folders automatically. Otherwise, your changes look great.

**Comment by puffnfresh** [comment]
@rahulmutt thanks, that was a problem. I had some more problems due to other ignored directories so I did a `git clean -fdx`. I also did a `rm -rf ~/.eta ~/.etlas` to be super clean.

I now have a working Eta compiler. This PR works well :fireworks:

File: default.nix [hunk 1]
```diff
@@ -21,11 +23,17 @@ let
...
 
+      # Tests rely on doctest which has problematic semigroups and binary
+      turtle = haskell.lib.dontCheck super.turtle;
+
       codec-jvm = self.callPackage ./utils/nix/codec-jvm.nix { };
       hackage-security = haskell.lib.dontCheck (self.callPackage ./utils/nix/hackage-security.nix { });
+      hpp = haskell.lib.dontCheck (haskell.lib.addExtraLibrary (self.callPackage ./utils/nix/hpp.nix { }) self.semigroups);
       eta-boot-meta = self.callPackage ./utils/nix/eta-boot-meta.nix { };
       eta-boot = self.callPackage ./utils/nix/eta-boot.nix { };
+      eta-meta = haskell.lib.dontHaddock (self.callPackage ./utils/nix/eta-meta.nix { });
```
Comment (puffnfresh) [inline default.nix]: I can get rid of this now that @rahulmutt has fixed the Haddock.

**Comment by rahulmutt** [comment]
@puffnfresh Everything looks good. Anything you want to change/add before I merge?

**Comment by puffnfresh** [comment]
@rahulmutt good to merge now. I removed the `dontHaddock` workaround.

**Comment by puffnfresh** [comment]
Also @jwiegley has tested out the expression and I think everything is working well for him.

**Pull request merged**